assessments/projects/cni/self-assessment.md

```diff
@@ -0,0 +1,274 @@
+# CNI Self-assessment
+
+This self-assessment was created as part of the Security Pals process.\
+Authors: Jason Lin, Quan Phung, Jackson Qu, Grace Liu
+
+## Table of contents
+
+* [Metadata](#metadata)
+  * [Security links](#security-links)
+* [Overview](#overview)
+  * [Actors](#actors)
+  * [Actions](#actions)
+  * [Background](#background)
+  * [Goals](#goals)
+  * [Non-goals](#non-goals)
+* [Self-assessment use](#self-assessment-use)
+* [Security functions and features](#security-functions-and-features)
+* [Project compliance](#project-compliance)
+* [Secure development practices](#secure-development-practices)
+* [Security issue resolution](#security-issue-resolution)
+* [Appendix](#appendix)
+
+## Metadata
+
+|   |  |
+| -- | -- |
+| Software | [A link to CNI's repository.](https://github.com/containernetworking/cni)  |
+| Security Provider | No |
+| Languages | Go |
+| SBOM | None generated by the CNI project at the moment |
+| | |
+
+### Security links
+
+| Doc | url |
+| -- | -- |
+| Security file | [README.md#Security](https://github.com/containernetworking/cni/blob/main/README.md#Security) |
+| Default and optional configs | [CNI Specification](https://www.cni.dev/docs/spec/) outlines required keys, optional keys, and protocol parameters |
+
+## Overview
+
+**Container Network Interface** <br>
+CNI, a project under the Cloud Native Computing Foundation, includes a set of guidelines and software libraries
+used for developing plugins that set up network interfaces within Linux and Windows containers. Its primary goal
+is managing container network connectivity and deallocating resources once a container is removed. This specific
+focus has led to broad support for CNI and a straightforward implementation process due to its simple specification.
+
+### Background
+
+With the rapid development of application containers on Linux, networking in this area is still not well addressed
+as it is highly environment-specific. To deal with this problem, developers seek to create a network layer that is
+pluggable for container runtimes. Ultimately, CNI, along with libraries for Go and a set of plugins, was introduced
```

Comment:
```suggestion
pluggable for container runtimes. Ultimately, CNI was introduced
```
Consider moving `along with libraries for Go and a set of plugins` to a sentence that can elaborate more specifically on what libraries and plugins you're referencing here.